tools/deviceGen/output/traits_defaults.hlsl

```diff
@@ -0,0 +1,382 @@
+NBL_GENERATE_GET_OR_DEFAULT(imageFootprint, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderIntegerFunctions2, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(pciDomain, uint32_t, ~0u);
+NBL_GENERATE_GET_OR_DEFAULT(pciBus, uint32_t, ~0u);
+NBL_GENERATE_GET_OR_DEFAULT(pciDevice, uint32_t, ~0u);
+NBL_GENERATE_GET_OR_DEFAULT(pciFunction, uint32_t, ~0u);
+NBL_GENERATE_GET_OR_DEFAULT(minFragmentDensityTexelSize, hlsl::uint32_t2, { ~0u, ~0u });
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentDensityTexelSize, hlsl::uint32_t2, { 0u, 0u });
+NBL_GENERATE_GET_OR_DEFAULT(fragmentDensityInvocations, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(decorateString, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderImageInt64Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(sparseImageInt64Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(lineSubPixelPrecisionBits, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat16Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat16AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat16AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat32AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat64AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat16Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat16AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat16AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat32AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat64AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderImageFloat32AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(sparseImageFloat32AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(deviceMemoryReport, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderNonSemanticInfo, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderEarlyAndLateFragmentTests, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(fragmentShaderBarycentric, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSubgroupUniformControlFlow, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(subsampledLoads, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(subsampledCoarseReconstructionEarlyAccess, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(maxSubsampledArrayLayers, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetSubsampledSamplers, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(workgroupMemoryExplicitLayout, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(workgroupMemoryExplicitLayoutScalarBlockLayout, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(workgroupMemoryExplicitLayout8BitAccess, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(workgroupMemoryExplicitLayout16BitAccess, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(colorWriteEnable, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(cooperativeMatrixSupportedStages, core::bitflag<asset::IShader::E_SHADER_STAGE>, asset::IShader::ESS_UNKNOWN);
```

Comment:
again no `bitflag` in HLSL (and even if you ported it it wouldn't be CONSTEXPR, so not usable as a default).

P.S. Also when you move `E_SHADER_STAGE` to `nbl::hlsl` you need to make it non `enum class`